addons/test_mail/tests/test_mail_message.py

```diff
@@ -373,3 +373,21 @@ def test_message_link_by_employee(self):
             self.assertEqual(res.url, expected_url)
         with self.subTest(deleted_message=deleted_message):
             res = self.url_open(f'/mail/message/{deleted_message.id}')
+
+@tagged("mail_message")
+class TestMessageFailure(MailCommon, HttpCase):
+
+    def test_failure_thread_display_name_as_name(self):
+        unnamed_record = self.env['mail.test.simple.unnamed'].create({'description': 'Some description'})
+        msg = unnamed_record.message_post(body='Some body', author_id=self.partner_employee.id)
+        self.env['mail.notification'].create({
+            'author_id': msg.author_id.id,
+            'mail_message_id': msg.id,
+            'res_partner_id': self.partner_employee.id,
+            'notification_type': 'email',
+            'notification_status': 'exception',
+            'failure_type': 'mail_email_invalid',
+        })
```

Comment:
better write a meaningful test case instead: post a message to a user with an invalid email, check it has a failure, check store results.

Reply:
I made this but there's still a need to make the `mail.notification` by hand to simulate the failure... Or you have an idea to prevent this and make it work?